wp-content/plugins/core/src/Templates/Components/Abstract_Controller.php

```diff
@@ -13,6 +13,15 @@
  */
 abstract class Abstract_Controller {
 
+	/**
+	 * @param array $args
+	 *
+	 * @return static
+	 */
+	public static function factory( array $args = [] ) {
```

Comment:
Looks like we don't get `:static` till PHP8. 😭  https://wiki.php.net/rfc/static_return_type

Reply:
I will just type it with the controller :)